plugin.go

```diff
@@ -28,40 +45,259 @@ func (p *Plugin) PluginInfo() launchr.PluginInfo {
+				}
+			} else {
+				// Store background outputs into plugin tmp folder.
+				err := redirectOutputs(pluginTmpDir)
+				if err != nil {
+					return err
+				}
+			}
+
+			if !server.IsPortFree(port) {
+				log.Info("web server port %d you are trying to use is not available", port)
+				port = server.GetFreePort(port, 20)
+			}
+
 			runOpts := &server.RunOptions{
-				Addr:        fmt.Sprintf(":%s", port), // @todo use proper addr
-				APIPrefix:   APIPrefix,
-				SwaggerJSON: useSwaggerUI,
-				ProxyClient: proxyClient,
+				Addr:          fmt.Sprintf(":%d", port), // @todo use proper addr
+				APIPrefix:     APIPrefix,
+				SwaggerJSON:   useSwaggerUI,
+				ProxyClient:   proxyClient,
+				RunInfoFolder: pluginTmpDir,
 				// @todo use embed fs for client or provide path ?
 			}
-
 			prepareRunOption(p, runOpts)
 
+			if foreground {
+				return server.Run(cmd.Context(), p.app, runOpts)
+			}
+
+			if !isBackGroundEnv() {
+				pid, err := p.runAsBackgroundCmd(cmd, pidFile)
+				if err != nil {
+					return err
+				}
+				// Wait until background server is up.
+				// Check if run info created and server is reachable.
+				// Print server URL in CLI.
+				// Kill process in case of timeout
+				ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+				defer cancel()
+
+				var wg sync.WaitGroup
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					for {
+						select {
+						case <-time.After(1 * time.Second):
+							runInfo, errRi := server.GetRunInfo(pluginTmpDir)
+							if errRi != nil {
+								continue
+							}
+
+							cli.Println("Web running in background with PID:%d", pid)
+							if runInfo.BaseURL != "" {
+								cli.Println(runInfo.BaseURL)
+							}
+							return
+						case <-ctx.Done():
+							return
+						}
+					}
+				}()
+
+				wg.Wait()
+
+				// Failed to retrieve run info for a reasonable time
+				if errCtx := ctx.Err(); errCtx != nil {
+					// Kill existing process
+					_ = killProcess(pid)
+
+					// Cleanup temp folder
+					err = os.RemoveAll(pluginTmpDir)
+					if err != nil {
+						log.Debug(err.Error())
+					}
+
+					return errors.New("couldn't start background process")
+				}
+
+				return nil
+			}
+
 			return server.Run(cmd.Context(), p.app, runOpts)
 		},
 	}
-	cmd.Flags().StringVarP(&port, "port", "p", "8080", `Web server port`)
+
+	cmd.Flags().IntVarP(&port, "port", "p", 8080, `Web server port`)
 	cmd.Flags().BoolVarP(&useSwaggerUI, "swagger-ui", "", false, `Serve swagger.json on /api/swagger.json and Swagger UI on /api/swagger-ui`)
+	cmd.Flags().BoolVarP(&foreground, "foreground", "", false, `Run server as foreground process`)
 	cmd.Flags().StringVarP(&proxyClient, "proxy-client", "", "", `Proxies to client web server, useful in local development`)
 	// Command flags.
 	rootCmd.AddCommand(cmd)
 	return nil
 }
+
+func (p *Plugin) runAsBackgroundCmd(cmd *cobra.Command, pidFile string) (int, error) {
```

Comment:
As mentioned above, maybe it's better to use https://github.com/sevlyar/go-daemon